### PR TITLE
Corn Integration

### PR DESCRIPTION
**Pull request by jalextowle**
# Description

Adds a `Corn` integration that makes it possible to launch a Hyperdrive pool with any of the assets allowed by the Corn Silo, adds an instance test for a Corn instance that uses LBTC, and adds deployment tools that deploy with LBTC.

### CODE REVIEW

File: codegen/example/config.yaml [hunk 1]
```diff
@@ -4,19 +4,16 @@
 name:
   # Capitalized version of the name to be used i.e. for contract names, file
   # names, and comments.
-  capitalized: "MorphoBlue"
+  capitalized: "Corn"
```
Comment (jrhea) [inline codegen/example/config.yaml:7]: did you mean to check this in?

File: contracts/src/instances/corn/CornBase.sol [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+    /// @dev Accepts a deposit from the user in base.
+    /// @param _baseAmount The base amount to deposit.
+    /// @return The shares that were minted in the deposit.
+    /// @return The amount of ETH to refund. Since this yield source isn't
+    ///         payable, this is always zero.
+    function _depositWithBase(
+        uint256 _baseAmount,
+        bytes calldata // unused
+    ) internal override returns (uint256, uint256) {
```
Comment (jrhea) [inline contracts/src/instances/corn/CornBase.sol:47]: i imagine this specific implementation is only for the lbtc silo, right? there is another silo that accepts dai, sdai, usdc so we'd need to account for the multiple base types for those silos, right?

File: tasks/deploy/config/mainnet/corn-lbtc-91day.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    prefix: "Corn",
+    coordinatorAddress: async (hre) =>
+        hre.hyperdriveDeploy.deployments.byName(MAINNET_CORN_COORDINATOR_NAME)
+            .address,
+    deploymentId: keccak256(toBytes(MAINNET_CORN_LBTC_91DAY_NAME)),
+    salt: toBytes32("0x69420"),
+    extraData: "0x",
+    contribution: CONTRIBUTION,
+    // FIXME
+    fixedAPR: parseEther("0.0666"),
```
Comment (jrhea) [inline tasks/deploy/config/mainnet/corn-lbtc-91day.ts:32]: the pendle pt is trading at 9.1%, fyi

File: contracts/src/instances/corn/CornBase.sol (hunk 2 above)
Comment (jalextowle) [reply contracts/src/instances/corn/CornBase.sol:47]: It should work for all of them. It’s made to be generic enough to handle all of the assets supported on the Corn Silo. For some assets, we’d need a zap, but I don’t think anything would need to change to support other assets.

File: contracts/src/instances/corn/CornHyperdrive.sol [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+/// |||////////000000000000\\\\\\\\|:::::::::::::::|////////00000000000\\\\\\\\\\|||
+/// SSS\\\\\\\\000000000000////////|:::::0x666:::::|\\\\\\\\00000000000//////////SSS
+/// SSS|||||||||||||||||||||||||||||:::::::::::::::||||||||||||||||||||||||||||||SSS
+/// SSSSSSSS|_______________|______________||_______________|______________|SSSSSSSS
+/// SSSSSSSS                                                                SSSSSSSS
+/// SSSSSSSS                                                                SSSSSSSS
+///
+/// @author DELV
+/// @title CornHyperdrive
+/// @notice A Hyperdrive instance that uses a Corn vault as the yield source.
```
Comment (mcclurejt) [inline contracts/src/instances/corn/CornHyperdrive.sol:54]: Might be nice to have a lil blurb here on what a corn silo is (assuming it's a vault)

**Review by mcclurejt — APPROVED** [review]
aMAIZEing

**Pull request merged**